src/browser/fixture.js

```diff
@@ -31,19 +31,41 @@ function getComposedChildren(node) {
 
 async function waitForElem(elem, awaitLoadingComplete = true) {
 
-	const update = elem?.updateComplete;
-	if (typeof update === 'object' && Promise.resolve(update) === update) {
-		await update;
-		await nextFrame();
-	}
+	const doWait = async() => {
 
-	if (awaitLoadingComplete && typeof elem?.getLoadingComplete === 'function') {
-		await elem.getLoadingComplete();
-		await nextFrame();
-	}
+		const update = elem?.updateComplete;
+		if (typeof update === 'object' && Promise.resolve(update) === update) {
+			await update;
+			await nextFrame();
+		}
+
+		if (awaitLoadingComplete && typeof elem?.getLoadingComplete === 'function') {
+			await elem.getLoadingComplete();
+			await nextFrame();
+		}
 
-	const children = getComposedChildren(elem);
-	await Promise.all(children.map(e => waitForElem(e, awaitLoadingComplete)));
+		const children = getComposedChildren(elem);
+		await Promise.all(children.map(e => waitForElem(e, awaitLoadingComplete)));
+
+	};
+
+	await new Promise((resolve) => {
+		const observer = new MutationObserver((records) => {
+			for (const record of records) {
+				for (const removedNode of record.removedNodes) {
+					if (removedNode === elem) {
+						observer.disconnect();
+						resolve();
+						return;
+					}
+				}
+			}
+		});
+		observer.observe(elem.parentNode, { childList: true });
+		doWait()
```

Comment:
Does this need to be returned as the promise so `waitForElem` doesn't resolve early? Actually not 100% sure how that works. That or use `await` here instead of `.then`?

Reply:
Never mind. This is down one layer, so that outer awaited promise will control `waitForElem`'s resolve timing.

Reply:
Yep. I would have rather put everything inside here, but you can't have an `async` executor (i.e. `new Promise(async(resolve) => { })` isn't allowed) so I had to use `.then` syntax in here.

Reply:
And by "you can't" I meant [you shouldn't](https://eslint.org/docs/latest/rules/no-async-promise-executor).